### ISSUE TITLE
Kodi a4kSubtitles Episode IMDB ID

### ISSUE THREAD

**Issue by woxie** [synclerd/syncler#25]
Syncler embeds the TV Show's IMDB ID and the season and episode number into the URL and sends it to Kodi via its webserver. It is ideal to use the episode's IMDB ID instead of the TV Show's IMDB ID for more accurate results.

Currently the URL looks like this:
`https://website.com/episode.mkv|ids={"tmdb":76669,"imdb":"tt7134908"}|season=2|episode=4`
[tt7134908](https://www.imdb.com/title/tt7134908/) = Elite

It should be like this:
`https://website.com/episode.mkv|ids={"tmdb":76669,"imdb":"tt9141176"}`
[tt9141176](https://www.imdb.com/title/tt9141176/) = Elite Season 2 Episode 4

With this change Syncler will be fully compatible with a4kSubtitles.

**Comment by sxiy** [comment]
a4kSubtitles was [updated a second time](https://github.com/a4k-openproject/a4kSubtitles/commit/36a30f54409a390b7e013af3cea37f7a2247c0df), to work with TV show IMDB ID + seasonNumber + episodeNumber.

Before this workaround update, searched subtitles would be for random episodes of that show, but I'm sure if you try with the latest a4kSubtitles version, you should only get subtitles for the episode you're watching.

Let us know if it's working as expected.

**Comment by woxie** [comment]
> a4kSubtitles was [updated a second time](https://github.com/a4k-openproject/a4kSubtitles/commit/36a30f54409a390b7e013af3cea37f7a2247c0df), to work with TV show IMDB ID + seasonNumber + episodeNumber.
> 
> Before this workaround update, searched subtitles would be for random episodes of that show, but I'm sure if you try with the latest a4kSubtitles version, you should only get subtitles for the episode you're watching.
> 
> Let us know if it's working as expected.

Yes, I did notice that the latest update addresses that, but unfortunately it's imperfect, unlike the alternative that I described above. A clear example of this is "The Offfice Season 4", which has five episodes separated into two parts.

- IMDB combines the split episodes, which gives a total of [14 episodes](https://www.imdb.com/title/tt0386676/episodes?season=4). 
- TMDB considers [19 episodes](https://www.themoviedb.org/tv/2316-the-office/season/4).
- Trakt considers [19 episodes](https://trakt.tv/shows/the-office/seasons/4).

This inconsistency between databases causes issues with the naming scheme and alters the results. For example, the episode titled "Night Out" is S04E11 on IMDB but S04E15 on Trakt/TMDB.

The results with Syncler's current URL format (`|ids={"tmdb":2316,"imdb":"tt0386676"}|season=4|episode=15`):
![image](https://user-images.githubusercontent.com/33608237/129987108-d92892b5-6576-4b1c-98e9-7ba5b7384aa6.png)

The results with the ideal format (`|ids={"tmdb":2316,"imdb":"tt1031450"}`)
![2021_08_18_19_01_52](https://user-images.githubusercontent.com/33608237/129987185-02051964-4b22-4009-b64f-6c89d33ef6e5.png)
![2021_08_18_19_02_00](https://user-images.githubusercontent.com/33608237/129987194-e107efcc-d5db-4f1c-9be2-c1bf2c9779a8.png)

And I haven't tested Specials, but they might not work reliably either.

**Comment by sxiy** [comment]
This makes total sense, thanks for the detailed report!

**Comment by woxie** [comment]
> This makes total sense, thanks for the detailed report!

No problem! However, I just tested the latest stable release and noticed that this wasn't modified, it still uses the same format. Was this request overlooked perhaps? This request was the [top comment](https://old.reddit.com/r/SynclerApp/comments/o5sdrk/syncler_13_rc_last_beta_before_stable_please_test/h2ojgr8/) on the first RC thread, which indicates that multiple users are interested in this change.

**Comment by Santaklawz** [comment]
I am interested in seeing this fixed too

**Comment by woxie** [comment]
> I am interested in seeing this fixed too

Any update on this? @sxiy
